packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -357,6 +370,22 @@ class _CupertinoSwitchState extends State<CupertinoSwitch> with TickerProviderSt
       ?? CupertinoColors.systemGreen,
       context,
     );
+    final (Color onLabelColor, Color offLabelColor)? onOffLabelColors =
+        MediaQuery.onOffSwitchLabelsOf(context)
+            ? (
+                CupertinoDynamicColor.resolve(
+                  widget.onLabelColor ?? CupertinoColors.white,
+                  context,
+                ),
+                CupertinoDynamicColor.resolve(
```

Comment:
Could you create a new `CupertinoDynamicColor` for this instead of putting the logic for handling high contrast here, and update the documentation for the default offLabelColor?

Reply:
I created a new `CupertinoDynamicColor` in `CupertinoColor`.

Is this ok, or should it not be in here `CupertinoColors` color as it's not reused in other widgets?

Reply:
Keeping it in `switch.dart` sgtm

Reply:
done